### ISSUE TITLE
Given/due/feedback for Homeworks not appropriate for peer-grading

### ISSUE THREAD

**Issue by florian-rabe** [slatex/sTeX-React#462]
When I create a homework (as opposed to a quiz), the process determined by the three dates of given-due-feedback is not practical.

Secondly, for ungraded/peer-graded homeworks, the following makes sense:
* one date for when the homework is doable relative to lecture progress
* after that data, submission and grading should be possible at any time

Is that something I can already accomplish with the current system?

**Comment by abchugh** [comment]
> given-due-feedback is not practical.
Can you elaborate a bit? What makes this impractical? 

> one date for when the homework is doable relative to lecture progress
Given the data that we have, it may be possible to do this but we need some ideas on how to make the notion of "relative to lecture progress" more concrete. The instructor specifying a "given date" was the obvious first step.

> after that data, submission and grading should be possible at any time
Grading is already possible at any time. Submission, on the other hand, is not.

We went with the process of a "due date" because it makes some things very convenient:
1. A deadline usually encourages students (and researchers 😄) to actually do the work. I suspect that an open deadline would result in fewer submissions.
2. This deadline results in a very simple logic about when the student's answer is considered final. They can keep updating their draft answer till the due date. 

If we want open submissions, an alternate solution would be to introduce the notion of a "draft answer vs final answer". Once the student marks an answer as "final", only then will it become available to instructors/peers for grading. At this point, we wouldn't allow students to update their answers because instructors/peers may have already started grading the answer.

**Comment by florian-rabe** [comment]
The current process makes sense for grade-relevant problems with official grading submissions deadlines, such as the quizzes.

But for practice problems (which includes the homeworks) and peer-grading, we need to rethink the process entirely.
Then submission must be possible at any time (e.g., after the lecture during exam prep), grading should be possible for whatever has been submitted, and feedback should be available for whatever has been graded.
So basically all three dates in the current homework creation dialog are useless.
Maybe you can make them optional, and if their omitted.
